src/Common/tests/Tests/System/IO/PathInternal.Tests.cs

```diff
@@ -64,5 +65,377 @@ public void GetCommonPathLength(string first, string second, bool ignoreCase, in
         {
             Assert.Equal(expected, PathInternal.GetCommonPathLength(first, second, ignoreCase));
         }
+
+        public static TheoryData<string, int, string> RemoveRelativeSegmentsData => new TheoryData<string, int, string>
+        {
+            { @"C:\git\corefx", 2, @"C:\git\corefx"},
+            { @"C:\\git\corefx", 2, @"C:\git\corefx"},
+            { @"C:\git\\corefx", 2, @"C:\git\corefx"},
+            { @"C:\git\.\corefx\.\\", 2, @"C:\git\corefx\"},
+            { @"C:\git\corefx", 2, @"C:\git\corefx"},
+            { @"C:\git\..\corefx", 2, @"C:\corefx"},
+            { @"C:\git\corefx\..\", 2, @"C:\git\"},
+            { @"C:\git\corefx\..\..\..\", 2, @"C:\"},
+            { @"C:\git\corefx\..\..\.\", 2, @"C:\"},
+            { @"C:\git\..\.\corefx\temp\..", 2, @"C:\corefx"},
+            { @"C:\git\..\\\.\..\corefx", 2, @"C:\corefx"},
+            { @"C:\git\corefx\", 2, @"C:\git\corefx\"},
+            { @"C:\git\temp\..\corefx\", 2, @"C:\git\corefx\"},
+
+            { @"C:\.", 3, @"C:\."},
+            { @"C:\..", 3, @"C:\.."},
+            { @"C:\..\..", 3, @"C:\"},
```

Comment:
`{ @"C:\\..\..", 3, @"C:\"},` would be interesting too. I'm ok with the output if it is `@"C:\\"`.

Reply:
the output is ```"C:\"```